lib/ansible/modules/cloud/vmware/vmware_content_library_info.py

```diff
@@ -0,0 +1,143 @@
+description:
+- Module to list the content libraries.
+- Module to get facts about specific content library
+- Content Library feature is introduced in vSphere 6.0 version, so this module is not supported in the earlier versions of vSphere.
+- All variables and VMware object names are case sensitive.
+version_added: '2.9'
+author:
+- Pavan Bidkar (@pgbidkar)
+notes:
+- Tested on vSphere 6.5, 6.7
+requirements:
+- python >= 2.6
+- PyVmomi
+- vSphere Automation SDK
+options:
+    library_id:
+      description:
+      - content library id for which details needs to be fetched.
+      type: str
+      required: False
+extends_documentation_fragment: vmware_rest_client.documentation
+'''
+
+EXAMPLES = r'''
+- name: Get List of Content Libraries
+  vmware_content_library_info:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+  delegate_to: localhost
+
+- name: Get facts about content library
+  vmware_content_library_info:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    library_id: '13b0f060-f4d3-4f84-b61f-0fe1b0c0a5a8'
+    validate_certs: no
+  delegate_to: localhost
+'''
+
+RETURN = r'''
+content_lib_details:
+  description: dictionary of content library metadata
+  returned: on success
+  type: dict
+  sample: {
+            "demo-local-lib": {
+            "lib_creation_time": "2019-07-02T11:50:52.242000",
+            "lib_description": "new description",
+            "lib_id": "13b0f060-f4d3-4f84-b61f-0fe1b0c0a5a8",
+            "lib_name": "demo-local-lib",
+            "lib_publish_info": {
+                "authentication_method": "NONE",
+                "persist_json_enabled": false,
+                "publish_url": null,
+                "published": false,
+                "user_name": null
+                },
+                "lib_server_guid": "0fd5813b-aac7-4b92-9fb7-f18f16565613",
+                "lib_type": "LOCAL",
+                "lib_version": "3"
+            }
+        }
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.vmware_rest_client import VmwareRestClient
+
+
+class VmwareContentLibFacts(VmwareRestClient):
+    def __init__(self, module):
+        """Constructor."""
+        super(VmwareContentLibFacts, self).__init__(module)
+        self.content_service = self.api_client
+        self.local_libraries = []
+        self.library_facts = dict()
+        self.lib_publish_info = dict()
+
+    def get_all_content_libs(self):
+        """Method to retrieve List of content libraries."""
+        self.module.exit_json(changed=False, content_libs=self.content_service.content.LocalLibrary.list())
+
+    def get_content_lib_details(self, library_id):
+        """Method to retrieve Details of contentlib with library_id"""
+        lib_details = self.content_service.content.LocalLibrary.get(library_id)
```

Comment:
I dont't have any resource and I get the following error:

```
Traceback (most recent call last):
  File "/home/goneri/.ansible/tmp/ansible-tmp-1562681655.5490832-209270742247135/AnsiballZ_vmware_content_library_info.py", line 139, in <module>
    _ansiballz_main()
  File "/home/goneri/.ansible/tmp/ansible-tmp-1562681655.5490832-209270742247135/AnsiballZ_vmware_content_library_info.py", line 131, in _ansiballz_main
    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
  File "/home/goneri/.ansible/tmp/ansible-tmp-1562681655.5490832-209270742247135/AnsiballZ_vmware_content_library_info.py", line 65, in invoke_module
    spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/tmp/ansible_vmware_content_library_info_payload_z7gmsr1d/__main__.py", line 143, in <module>
  File "/tmp/ansible_vmware_content_library_info_payload_z7gmsr1d/__main__.py", line 137, in main
  File "/tmp/ansible_vmware_content_library_info_payload_z7gmsr1d/__main__.py", line 105, in get_content_lib_details
  File "/home/goneri/.virtualenvs/ansible/lib/python3.7/site-packages/com/vmware/content_client.py", line 744, in get
    'library_id': library_id,
  File "/home/goneri/.virtualenvs/ansible/lib/python3.7/site-packages/vmware/vapi/bindings/stub.py", line 317, in _invoke
    return self._api_interface.native_invoke(ctx, _method_name, kwargs)
  File "/home/goneri/.virtualenvs/ansible/lib/python3.7/site-packages/vmware/vapi/bindings/stub.py", line 275, in native_invoke
    self._rest_converter_mode)
com.vmware.vapi.std.errors_client.NotFound: {messages : [LocalizableMessage(id='com.vmware.vdcs.cls-main.validate_id_not_found', default_message='ID content_library of resource type [com.vmware.content.Library] not found.', args=['content_library', 'com.vmware.content.Library'])], data : None}
```

I think it would be more Ansible-ish to return an explicity `msg` key. This way, the users won't have to set `ignore_errors`.

Reply:
Made the change. Now error message looks like
```
TASK [Get facts about content Library] *************************************************************************task path: /home/vmware/Ansible/playbooks/content_libs/list_content_libs.yml:6
fatal: [localhost]: FAILED! => {"changed": false, "exists": false, "msg": "Failed to find the library 13b0f060-f4d3-4f84-b61f-0fe1b0c0a5a8"}
```